picard/ui/searchdialog.py

```diff
@@ -452,11 +457,258 @@ def load_selection(self, row):
+        if size:
+            self.resize(size)
+        self.search_box.restore_checkbox_state()
+
+    def restore_table_header_state(self):
+        header = self.table.horizontalHeader()
+        state = config.persist["tracksearchdialog_header_state"]
+        if state:
+            header.restoreState(state)
+        header.setResizeMode(QtGui.QHeaderView.Interactive)
+
+    def save_state(self):
+        if self.table:
+            self.save_table_header_state()
+        config.persist["tracksearchdialog_window_size"] = self.size()
+
+    def save_table_header_state(self):
+        state = self.table.horizontalHeader().saveState()
+        config.persist["tracksearchdialog_header_state"] = state
+
+
+class AlbumSearchDialog(SearchDialog):
+
+    options = [
+        config.Option("persist", "albumsearchdialog_window_size", QtCore.QSize(720, 360)),
+        config.Option("persist", "albumsearchdialog_header_state", QtCore.QByteArray())
+    ]
+
+    def __init__(self, parent):
+        super(AlbumSearchDialog, self).__init__(parent)
+        self.cluster = None
+        self.setWindowTitle(_("Album Search Results"))
+        self.table_headers = [
+                _("Name"),
+                _("Artist"),
+                _("Format"),
+                _("Tracks"),
+                _("Date"),
+                _("Country"),
+                _("Labels"),
+                _("Catalog #s"),
+                _("Barcode"),
+                _("Language"),
+                _("Type"),
+                _("Status"),
+                _("Cover")
+        ]
+
+    _coverart_column = 12
+
+    def search(self, text):
+        """Perform search using query provided by the user."""
+        self.retry_params = Retry(self.search, text)
+        self.search_box.search_edit.setText(text)
+        self.show_progress()
+        self.tagger.xmlws.find_releases(self.handle_reply,
+                query=text,
+                search=True,
+                limit=25)
+
+    def show_similar_albums(self, cluster):
+        """Perform search by using existing metadata information
+        from the cluster as query."""
+        self.retry_params = Retry(self.show_similar_albums, cluster)
+        self.cluster = cluster
+        metadata = cluster.metadata
+        query = {
+            "artist": metadata["albumartist"],
+            "release": metadata["album"],
+            "tracks": str(len(cluster.files))
+        }
+
+        # Generate query to be displayed to the user (in search box).
+        # If advanced query syntax setting is enabled by user, display query in
+        # advanced syntax style. Otherwise display only album title.
+        if config.setting["use_adv_search_syntax"]:
+            query_str = ' '.join(['%s:(%s)' % (item, escape_lucene_query(value))
+                                for item, value in query.iteritems() if value])
+        else:
+            query_str = query["release"]
+
+        query["limit"] = 25
```

Comment:
I would put this `25` in a constant, at top of this file (if not used elsewhere).